### ISSUE TITLE
Update to the yellow highlight when existing a thread

### ISSUE THREAD

**Issue by sonialiap** [Expensify/App#42165]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:**
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:** n/a
**Email or phone of affected tester (no customers):** everyone
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** @JmillsExpensify 
**Slack conversation:** https://expensify.slack.com/archives/C03U7DCU4/p1714062455602839

## Action Performed:
1. Click into a thread
2. Click back into the main conversation by clicking the the room title in the top left of the thread window
3. Observe that the thread is highlighted in yellow after you exit it
## Expected Result:
The yellow highlight should fade after
- Clicking out of the chat into another chat
- Clicking into another thread
- Refreshing the page
- Sending a new comment in the same chat (but not receiving a comment from another user)
- Taking any action from the + menu while in the chat

## Actual Result:
Highlight remains until you click out of the chat into another chat. It does not clear after any of the other cases mentioned above.
## Workaround:
Usable but many have found the highlight confusing and distracting
## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos
https://github.com/Expensify/App/assets/17131195/aa847e2d-cda1-40ba-a825-f3e1affc5129

<img width="1141" alt="image" src="https://github.com/Expensify/App/assets/17131195/aad0bdfd-29f1-4ada-af3a-2803d7fc221b">

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by brunovjk** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
The yellow thread highlight in the Expensify app persists even after the user has exited the thread and performed other actions within the same chat. 

### What is the root cause of that problem?
The root cause lies in the `highlightedBackgroundColorIfNeeded` logic within `ReportActionItem.tsx`. Currently, the highlight is only cleared when the user navigates to a different chat. It does not consider other actions that should logically dismiss the highlight.

### What changes do you think we should make in order to solve the problem?
We should modify the `highlightedBackgroundColorIfNeeded` logic to include additional conditions for clearing the highlight. Specifically, the highlight should be cleared when:
- The user clicks out of the chat into another chat.
- The user clicks into another thread within the same chat.
- The user refreshes the page.
- The user sends a new comment in the same chat.
- The user takes any action from the "+" menu while in the chat.

To achieve this, we can introduce a state variable (e.g., `isThreadHighlighted`) and a `useEffect` hook to manage the highlight state. The hook would listen for the relevant events (e.g., navigation changes, new comments, "+" menu actions) and update the `isThreadHighlighted` state accordingly. The `highlightedBackgroundColorIfNeeded` logic would then depend on this state variable to determine whether to apply the highlight.

### What alternative solutions did you explore? (Optional)
One alternative solution could be to use a timer to automatically clear the highlight after a certain delay. However, this approach might not be ideal as it could lead to inconsistent behavior depending on the user's actions. Another alternative is to clear the highlight whenever the user interacts with any element outside the highlighted thread. This might be too broad and could unintentionally clear the highlight in some cases.

**Comment by nkdengineer** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Highlight remains until you click out of the chat into another chat. It does not clear after any of the other cases mentioned above.

### What is the root cause of that problem?
The report action is always highlighted if it's the linked action.


https://github.com/Expensify/App/blob/edd1719f62d74663d52f4168db2fd7adfe0cdda3/src/pages/home/report/ReportActionItem.tsx#L215
### What changes do you think we should make in order to solve the problem?
I think we can do the same behavior in Slack that only highlights the link action in a few seconds so we don't need to take care what action we should clear the highlighted action

We can do this by using animated view here

https://github.com/Expensify/App/blob/edd1719f62d74663d52f4168db2fd7adfe0cdda3/src/pages/home/report/ReportActionItem.tsx#L951

1. Create a shared value 

```
const bgColor = useSharedValue<string>(isReportActionLinked ? theme.messageHighlightBG : 'inherit');
```

2. Create a useEffect to change the value to inherit


```
useEffect(() => {
    if (!isReportActionLinked) {
        return;
    }

    bgColor.value = withTiming('inherit', {
        duration: 3000,
        easing: Easing.inOut(Easing.ease),
    })
}, [isReportActionLinked, theme.messageHighlightBG])
```

3. Create a useEffect to do the animation again if the page is changed from un-focused to focused

```
const isFocused = useIsFocused();
const prevIsFocused = usePrevious(isFocused);
useEffect(() => {
    if (!isFocused || prevIsFocused || !isReportActionLinked) {
        return;
    }

    bgColor.value = theme.messageHighlightBG;
    bgColor.value = withTiming('inherit', {
        duration: 3000,
        easing: Easing.inOut(Easing.ease),
    })
}, [isFocused, prevIsFocused, isReportActionLinked])
```

4. Change the style here to use animated style

```
const highlightedBackgroundColorIfNeeded = useAnimatedStyle(() =>
    StyleUtils.getBackgroundColorStyle(bgColor.value)
);
```

https://github.com/nkdengineer/App/blob/edd1719f62d74663d52f4168db2fd7adfe0cdda3/src/pages/home/report/ReportActionItem.tsx#L215

5. Use animated view here

```
<Animated.View style={highlightedBackgroundColorIfNeeded}>
```

https://github.com/Expensify/App/blob/edd1719f62d74663d52f4168db2fd7adfe0cdda3/src/pages/home/report/ReportActionItem.tsx#L951

### What alternative solutions did you explore? (Optional)
NA


### Result


https://github.com/Expensify/App/assets/161821005/203242b6-ba90-498d-b35f-697a96a75969

**Comment by lschurr** [comment]
Hi @sobitneupane - Can you take a look at these proposals?

**Comment by sobitneupane** [comment]
Thanks for the [proposal](https://github.com/Expensify/App/issues/42165#issuecomment-2111017989) @brunovjk 

> The hook would listen for the relevant events (e.g., navigation changes, new comments, "+" menu actions) and update the isThreadHighlighted state accordingly.

Can you please add more details about each events that hook would listen to and how will you handle it?

**Comment by brunovjk** [comment]
Thanks for the review @sobitneupane

## Proposal
[Updated](https://github.com/Expensify/App/issues/42165#issuecomment-2111017989)

Added more detailed implementation steps for handling events that should clear the thread highlight

**Comment by sobitneupane** [comment]
Thanks for the update @brunovjk 

[Proposal](https://github.com/Expensify/App/issues/42165#issuecomment-2111017989) from @brunovjk looks good to me.

🎀 👀 🎀 C+ reviewed

**Comment by nkdengineer** [comment]
> Clicking out of the chat into another chat
Clicking into another thread
Refreshing the page
Sending a new comment in the same chat (but not receiving a comment from another user)
Taking any action from the + menu while in the chat

@sobitneupane I think we shouldn't removing the highlight of a message when we do these action because in addition to those, we have many other actions that should also remove the highlight like edit another comment, react, ... Instead, we should only have an animation for linked report action like Slack so we don't need to care what action we should clear the highlight. What do you think?

cc @yuwenmemon

**Comment by sobitneupane** [comment]
> I think we shouldn't removing the highlight of a message when we do these action because in addition to those, we have many other actions that should also remove the highlight like edit another comment, react, ... Instead, we should only have an animation for linked report action like Slack so we don't need to care what action we should clear the highlight.

That's a good point @nkdengineer. I kind of agree with you. @yuwenmemon What's your take on highlighting just for few seconds and fading it out?

**Comment by brunovjk** [comment]
I also think it makes sense, I even pointed out this solution in the "alternative solutions" section before

**Comment by brunovjk** [comment]
I also think it makes sense, I even pointed out this solution in the "alternative solutions" section before. We just need to test it carefully, but I think it will be good too.

**Comment by brunovjk** [comment]
What do you think about using both options together? It disappeared after a while and/or when the user performed some action.

**Comment by yuwenmemon** [comment]
Hmmm... I'd defer to @Expensify/design on this question.

The question is: Should the "yellow highlight" that highlights a linked message disappear a few seconds after showing?

**Comment by shawnborton** [comment]
I think we want to do what is outlined in the original comment:

> The yellow highlight should fade after
> 
> Clicking out of the chat into another chat
> Clicking into another thread
> Refreshing the page
> Sending a new comment in the same chat (but not receiving a comment from another user)
> Taking any action from the + menu while in the chat

cc @sonialiap for confirmation though. I know this one was a long discussion, and I'm fairly certain we landed on the scenarios listed above.

**Comment by dannymcclain** [comment]
>The question is: Should the "yellow highlight" that highlights a linked message disappear a few seconds after showing?

This has come up several times in Slack and there's been quite a bit of discussion about it. We've consistently received push-back about the highlight fading after a set period of time (some people are for it, some people are against it), and I think the consensus was to rely on actions to dismiss it rather than time. This seemed to be the best compromise amongst the entire team, so I think we should continue to pursue that path.

**Comment by shawnborton** [comment]
Exactly ^ 🤣

**Comment by brunovjk** [comment]
Awesome! @yuwenmemon  Can I move forward with my [proposal](https://github.com/Expensify/App/issues/42165#issuecomment-2111017989)?

**Comment by yuwenmemon** [comment]
Yep

**Comment by brunovjk** [comment]
@yuwenmemon I can complete the task within a week. Is there a target deadline, is it needed sooner?
Thank you!

**Comment by brunovjk** [comment]
ETA 05/30

**Comment by brunovjk** [comment]
Not Overdue. Continue working in PR

**Comment by lschurr** [comment]
Go away Melv - https://github.com/Expensify/App/issues/42165#issuecomment-2135829854

**Comment by brunovjk** [comment]
@lschurr, @yuwenmemon and @sobitneupane I found the task more challenging than expected. ETA might need to extend. Will keep working and raise the PR ASAP. Is this okay? Thank you.

**Comment by yuwenmemon** [comment]
All good

**Comment by brunovjk** [comment]
**Update**: Keep working on the PR, I will raise it ASAP

**Comment by lschurr** [comment]
Any update @brunovjk?

**Comment by brunovjk** [comment]
Sorry for the delay, I had to focus on another issue but I'm back here, I just need to do a few tests and we're ready :D

**Comment by brunovjk** [comment]
**Update**: Keep working on the PR

**Comment by brunovjk** [comment]
**Update**: Keep working on the PR

**Comment by brunovjk** [comment]
**Update:** Same as above

**Comment by brunovjk** [comment]
**Update:** Same as above

**Comment by brunovjk** [comment]
**Update**: I got almost all cover:
- [x] - Clicking out of the chat into another chat
- [x] - Clicking into another thread
- [ ] - Refreshing the page
- [x] - Sending a new comment in the same chat (but not receiving a comment from another user)
- [x] - Taking any action from the + menu while in the chat

I need to **confirm the expected behavior for the highlight when refreshing the page**. According to the previous [PR](https://github.com/Expensify/App/pull/27227) that introduced the highlight for a linked report, a report item should be highlighted when a user sends a link (e.g., `456456456456/r/465456456464`) to another user. However, **if the user refreshes the page using the same link, should the highlight still be removed?**

**Comment by dannymcclain** [comment]
>However, if the user refreshes the page using the same link, should the highlight still be removed?

I _believe_ this is the desired behavior—remove the highlight if you refresh the page. I'll let others chime in too to make sure I've followed the conversation correctly.

**Comment by brunovjk** [comment]
Melvin PR is almost ready

**Comment by brunovjk** [comment]
**Update**: Still need to safely implement the dismissal when the user refreshes the page, without crashing the new 'Comment Linking' feature.

**Comment by brunovjk** [comment]
I'm stuck on **clearing the highlight comment** (`linkedReportActionID`) **during a page refresh**. We need to call `resetLinkedReportActionID()` (created in our [PR](https://github.com/Expensify/App/pull/43581)) only on page refresh, not on deep link navigation or any other way.
@sobitneupane any suggestions on **differentiating between a page refresh and accessing via a link**? Thanks!

**Comment by brunovjk** [comment]
**Update**: Testing on multiple platforms.
**Next**: Record videos and raise PR.

**Comment by brunovjk** [comment]
**Update:** Same as above

**Comment by brunovjk** [comment]
**Update**: Continue working on PR.

**Comment by brunovjk** [comment]
**Update**: Continue working on PR. Catching possible regressions.

**Comment by brunovjk** [comment]
**Update:** Same as above

**Comment by brunovjk** [comment]
**Update**: The feat to reset the yellow highlight when refreshing a page is taking a little more work than expected, but going well, get back with the PR to review ASAP

**Comment by brunovjk** [comment]
**Update:** Same as above

**Comment by brunovjk** [comment]
**Update**: Continue working on PR. Catching possible regressions.

**Comment by sobitneupane** [comment]
@shawnborton @yuwenmemon Should we remove the yellow highlight as soon as we begin taking any action (after clicking Submit expense from the + menu), or after the action is complete?

**Comment by shawnborton** [comment]
Probably as soon as we begin taking any action? Curious what @Expensify/design and @yuwenmemon thinks though.

**Comment by dannymcclain** [comment]
>Probably as soon as we begin taking any action?

Makes sense to me.

**Comment by dubielzyk-expensify** [comment]
Agree with the designers

**Comment by brunovjk** [comment]
**Update**: PR in review

**Comment by brunovjk** [comment]
**Update**: Continue working on PR. Catching possible regressions.

**Comment by brunovjk** [comment]
Same as above.

**Comment by brunovjk** [comment]
Hi team @yuwenmemon @sobitneupane,

I've been working on the [yellow highlight behavior update](https://github.com/Expensify/App/pull/43581) and initially pursued a more robust solution by resetting `linkedReportActionID` using `Navigation.setParams({reportActionID: ''})`, [since the highlight item depends on the `linkedReportActionID`](https://github.com/Expensify/App/blob/01c4d421f61cad6a8ff7810a809e1b70a13f878f/src/pages/home/report/ReportActionItem.tsx#L200-L209).
This successfully removes the highlight but causes unwanted scrolling when the highlighted message is above the chat's end. After several attempts, including different strategies, I reverted to the original [proposal](https://github.com/Expensify/App/issues/42165#issuecomment-2111017989) of managing the highlight via state. However, this approach introduced regressions, especially with refreshing the page. I faced significant challenges in detecting page refreshes, specifically on web, and implementing a reliable system to clear the highlight without affecting other functionalities.

Despite daily efforts over months, I couldn't find a solution that avoids breaking the current behavior. Hence, I request to be unassigned from this task, as I believe someone else might have better success. Is there a problem?

I sincerely apologize for any delays and remain available to assist during the transition until this feature works flawlessly.

Thank you for your understanding.

**Comment by yuwenmemon** [comment]
Yes, no worries at all @brunovjk. Let's reopen.

**Comment by brunovjk** [comment]
Thank you all for your understanding.

**Comment by sobitneupane** [comment]
Thanks for your work so far, @ishpaul777. I'm open to the idea and look forward to seeing your proposal.

**Comment by wildan-m** [comment]
Is the expected behavior only when exiting a thread?

What if the user deeplink from report action copy link? should it behave the same?

**Comment by lschurr** [comment]
What's next for this one @sobitneupane? Are we waiting on a proposal from @ishpaul777?

**Comment by wildan-m** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Update the yellow thread highlight behavior upon exiting a thread.

### What is the root cause of that problem?

New feature

### What changes do you think we should make in order to solve the problem?

Since there should be specific logic when re-load we'll need to store highlighted action in persist storage, e.g. onyx.

Introduce new onyx key `ACTION_HIGHLIGHT`

```tsx

type ActionHighlight = {
    reportActionID: string;
    isHighlighted: boolean;
    isVisited: boolean;
};

   [ONYXKEYS.ACTION_HIGHLIGHT]: ActionHighlight;

```

Then we would need function to dismiss and reset the highlight, it can be in `ReportActionUtils` or a separate file:

<details><summary>dismiss and reset function</summary>

```tsx

function dismissHighlight(shouldUseSet = false)
{
    if(shouldUseSet)
    {
        if (!currentActionHighlight || !currentActionHighlight.reportActionID || !currentActionHighlight.isHighlighted)
        {
            return;
        }
        Onyx.set(ONYXKEYS.ACTION_HIGHLIGHT, {reportActionID: currentActionHighlight?.reportActionID, isVisited: currentActionHighlight?.isVisited, isHighlighted: false});
        return;
    }
    // use merge by default to avoid unnecessary render in useEffect
    Onyx.merge(ONYXKEYS.ACTION_HIGHLIGHT, {isHighlighted: false});
}

function resetHighlight()
{
    Onyx.set(ONYXKEYS.ACTION_HIGHLIGHT, null);
}

```

</details>

Then we have to modify highlight logic:

- `isFocused` is used for re-highlight the action when we navigate to parent report thread
- `actionHighlight.isVisited` is to track if the highlight already visited and we'll dismiss highlight if we reload the page and `isVisited` is true.
- `actionHighlight.reportActionID` is to know which report action is currently active

<details><summary>Modify ReportAction logic useEffect</summary>

```tsx

    const isFocused = useIsFocused();
    
    useEffect(() => {
        if (!linkedReportActionID || linkedReportActionID === '-1') {
            console.log("[wildebug] linkedReportActionID is not valid:", linkedReportActionID);
            return;
        }
    
        if (!action.reportActionID || !isFocused) {
            console.log("[wildebug] action.reportActionID or isFocused is not valid");
            return;
        }
    
        const isReportActionLinked = linkedReportActionID === action.reportActionID;
        console.log("[wildebug] isReportActionLinked:", isReportActionLinked);
    
        if (!isReportActionLinked) {
            console.log("[wildebug] isReportActionLinked is false");
            return;
        }
    
        if (actionHighlight?.reportActionID === action.reportActionID) {
            console.log("[wildebug] actionHighlight matches action.reportActionID");
            if (actionHighlight.isVisited) {
                console.log("[wildebug] actionHighlight isVisited is true, setting isHighlighted to false");
                Onyx.merge(ONYXKEYS.ACTION_HIGHLIGHT, { isHighlighted: false });
            } else {
                console.log("[wildebug] actionHighlight isVisited is false, setting isVisited to true");
                Onyx.merge(ONYXKEYS.ACTION_HIGHLIGHT, { isVisited: true });
            }
            return;
        }
    
        console.log("[wildebug] Initializing or replacing actionHighlight");
        Onyx.merge(ONYXKEYS.ACTION_HIGHLIGHT, { reportActionID: action.reportActionID, isHighlighted: true, isVisited: false });
    }, [actionHighlight?.reportActionID, linkedReportActionID, action.reportActionID, isFocused]);
   

```

</details>

Then Modify highlightedBackgroundColorIfNeeded logic to use our persisted highlight as main source of value to determine the highlighting.

<details><summary>Modify highlightedBackgroundColorIfNeeded logic</summary>

```tsx

    const isReportActionLinked = linkedReportActionID && action.reportActionID && linkedReportActionID === action.reportActionID;
    const highlightedBackgroundColorIfNeeded = useMemo(() => {
        const isHighlighted = actionHighlight?.reportActionID && action.reportActionID && actionHighlight?.isHighlighted && actionHighlight.reportActionID === action.reportActionID;
        if ((isHighlighted) {
            return StyleUtils.getBackgroundColorStyle(theme.messageHighlightBG);
        }
        return {};
    }, [StyleUtils, actionHighlight?.reportActionID, actionHighlight?.isHighlighted, action.reportActionID, theme.messageHighlightBG]);


```

</details>

To avoid highlight flickering when highlighting last report action of a chat we can listen `beforeunload` to dismiss highlight before we reload. We should use Onyx.set instead of merge since the page reloading can cancel `Onyx.merge`

<details><summary>Resolve highlight flickering</summary>

ReportActionsList.tsx
```tsx

    useEffect(() => {
        if(linkedReportActionID === '-1')
        {
            // no report action id defined in param, reset highlight
            ReportActionsUtils.resetHighlight();
            return;
        }

        const dismissActionHighlight = () => {
            ReportActionsUtils.dismissHighlight(true);
        };

        window.addEventListener('beforeunload', dismissActionHighlight);

        return () => {
            window.removeEventListener('beforeunload', dismissActionHighlight);
        };
    }, [linkedReportActionID]);

```

</details>

**Optional**
If linking from `Copy Link` context menu should behave differently (e.g. it should persist highlight and not affected by user action dismissal), then we can distinguish the behavior by using query string. We can use existing `referrer` param or use separate param to determine `highlight-mode`

CONST.TS
```tsx

    REFERRER: {
        NOTIFICATION: 'notification',
        COPY_LINK: 'copy-link',  < --- add new referer
    },


```

Then we can modify the highlight logic to not affected by the user action

```tsx

    const isFromCopyLink = route?.params?.referrer === CONST.REFERRER.COPY_LINK;
    const highlightedBackgroundColorIfNeeded = useMemo(() => {
        const isHighlighted = actionHighlight?.reportActionID && action.reportActionID && actionHighlight?.isHighlighted && actionHighlight.reportActionID === action.reportActionID;
        if ((isHighlighted || (isReportActionLinked && isFromCopyLink))) {
            return StyleUtils.getBackgroundColorStyle(theme.messageHighlightBG);
        }
        return {};
    }, [StyleUtils, actionHighlight?.reportActionID, actionHighlight?.isHighlighted, action.reportActionID, theme.messageHighlightBG]);


```

**Summary**

- Clicking out of the chat into another chat (Call dimsissHighlight in ReportActionItem onPress)
- Clicking into another thread (Call resetHighlight when navigating to parent report)
- Refreshing the page (Utilize `isVisited` to determine `isHighlighted`, dismiss the highlight `beforeunload`)
- Sending a new comment in the same chat  (Call dimsissHighlight in Composer submission events)
- Taking any action from the + menu while in the chat (Call dimsissHighlight in menu onSelect)

[Branch for this solution](https://github.com/Expensify/App/compare/main...wildan-m:wildan/fix/42165-action-base-highlight-dismiss?expand=1)
### What alternative solutions did you explore? (Optional)
N/A

**Comment by lschurr** [comment]
Could you review @sobitneupane?

**Comment by sobitneupane** [comment]
@wildan-m @ishpaul777 Could you please confirm that the highlight gets removed on taking all of the following actions?

>- Clicking out of the chat into another chat
>- Clicking into another thread
>- Refreshing the page
>- Sending a new comment in the same chat (but not receiving a comment from another user)
>- Taking any action from the + menu while in the chat

**Comment by sobitneupane** [comment]
Thanks for your proposal @wildan-m  I think persistent storage might not be ideal since the highlight is temporary and should reset when users switch chats, refresh, or take [these](https://github.com/Expensify/App/issues/42165#issuecomment-2422202212) actions. We only want this behavior for the current session.

**Comment by wildan-m** [comment]
I can confirm that my solution can resolve all cases, but not implemented all action in the POC, especially the last requirement


without persistent storage how can we distinguish when user re-visit the same report linking or a different one, AFAIK global var / context will reset on each browser refresh

**Comment by sobitneupane** [comment]
@ishpaul777 We would like the highlight to be removed in each of the cases mentioned [here](https://github.com/Expensify/App/issues/42165#issuecomment-2422202212).

**Comment by dannymcclain** [comment]
Honestly I think if we've taken care of every other case, I don't mind leaving the highlight on refresh. It's the least weird / lowest priority IMO.

**Comment by sobitneupane** [comment]
Thanks @dannymcclain for chiming in.

> Honestly I think if we've taken care of every other case, I don't mind leaving the highlight on refresh. It's the least weird / lowest priority IMO.

@brunovjk I believe your PR handled every other case except refreshing. I believe we can proceed with your PR in that case. Can you please confirm?

**Comment by brunovjk** [comment]
Hi @sobitneupane, thanks for asking! And yes! In [our PR](https://github.com/Expensify/App/pull/43581/files), we address:
- Sending a new comment in the same chat (but not receiving a comment from another user)
- Taking any action from the + menu while in the chat

Note: _Clicking out of the chat into another chat_ and _Clicking into another thread_ already dismiss/update the highlight, since in both scenarios we update the `linkedReportActionID`.

My solution offers simplicity and component-level control. However @ishpaul777's [proposal](https://github.com/Expensify/App/issues/42165#issuecomment-2408656910) could be more robust in a global context, but it might be slightly outside the original issue’s scope. I’m happy to move forward with [my solution](https://github.com/Expensify/App/pull/43581/files) if it fits the issue’s needs, but if you decide to go with @ishpaul777's proposal I also think it's a good idea and I’m available to help with testing and implementation, regardless of compensation.

**Comment by sobitneupane** [comment]
Thanks @brunovjk . I appreciate your understanding.

[Proposal](https://github.com/Expensify/App/issues/42165#issuecomment-2408656910) from @ishpaul777 looks good to me.  I believe it will make the thread highlight behavior more consistent and intuitive.

cc: @yuwenmemon

**Comment by twisterdotcom** [comment]
Relabelling as Lauren is OOO now.

**Comment by sobitneupane** [comment]
PR in review.

**Comment by sobitneupane** [comment]
Thanks for your effort, @ishpaul777!

@yuwenmemon, this task has already been unassigned by two contributors. Should we open it up for contributors again, or handle it through the agency?

**Comment by brunovjk** [comment]
Since we shouldn't include "refresh page" I believe my solution can work. I'll create a PR draft now, it will be ready by tomorrow and we can test it. If not I think it's a good idea to leave it to an agency to handle. Thanks.

**Comment by brunovjk** [comment]
@sobitneupane I made a draft PR [here](https://github.com/Expensify/App/pull/54874). The initial task of _The yellow highlight should fade after_:

1. Clicking out of the chat into another chat
2. Clicking into another thread
3. Sending a new comment in the same chat (but not receiving a comment from another user)
4. Taking any action from the + menu while in the chat

**is working as expected on my end**. I now need to perform more extensive testing on the app to catch any bugs we may have introduced. During this, and before opening it for a full review, could you provide feedback on our approach and typing in [`clearReportAction.ts`](https://github.com/Expensify/App/blob/eaa736b90e0d29f598b601940838bba4143ea9b3/src/pages/home/report/clearReportAction.ts) and its integration in `ReportActionsList.tsx`? 

Thank you very much for your time.

**Comment by brunovjk** [comment]
I'm working on it

**Comment by sobitneupane** [comment]
@yuwenmemon  We've attempted to resolve this issue through multiple PRs, but ultimately, all of them were dropped. I'd like to suggest handling it through agency. However, I'm open to reopening it for external contributors if that's preferred.
